modules/connectors/ups/karrio/providers/ups/rate.py

```diff
@@ -27,35 +28,35 @@ def _extract_details(
     ctx: dict,
 ) -> typing.List[models.RateDetails]:
     rate = lib.to_object(ups_response.RatedShipmentType, detail)
-    effective_rate = rate.NegotiatedRateCharges if rate.NegotiatedRateCharges is not None else rate
+    effective_rate = lib.identity(
+        rate.NegotiatedRateCharges if rate.NegotiatedRateCharges is not None else rate
+    )
     total_charge = effective_rate.TotalChargesWithTaxes or effective_rate.TotalCharges
     taxes = effective_rate.TaxCharges or []
     itemized_charges = [*effective_rate.ItemizedCharges, *taxes]
 
     charges = [
         ("Base charge", effective_rate.BaseServiceCharge.MonetaryValue),
-        *(
+        *lib.identity(
             []
             if any(itemized_charges)
             else [("Taxes", sum(lib.to_money(c.MonetaryValue) for c in taxes))]
         ),
-        *(
+        *lib.identity(
             (rate.Service.Code, rate.ServiceOptionsCharges.MonetaryValue)
             if lib.to_int(rate.ServiceOptionsCharges.MonetaryValue) > 0
             else []
         ),
-        *(
+        *lib.identity(
             (
-                (
+                lib.identity(
                     provider_units.SurchargeType.map(
                         str(getattr(c, "Code", None) or getattr(c, "Type", None))
-                    ).name.capitalize().replace("_", " ")
-                    if provider_units.SurchargeType.map(
-                        str(getattr(c, "Code", None) or getattr(c, "Type", None))
-                    ).name
-                    else (getattr(c, "Code", None) or getattr(c, "Type", None))
+                    )
+                    .name_or_key.replace("_", " ")
+                    .upper()
```

Comment:
I was using `capitalize` so it matches `Base charge`

Reply:
also i see that `upper` has `.replace("_", " ")` already in it

Reply:
The issue was that things like HST would ended up being `Hst` so to make it uniform that I replace to upper.
I will also make `Base Charge` upper to keep all uniform.

Reply:
> The issue was that things like HST would ended up being Hst so to make it uniform that I replace to upper.
I will also make Base Charge upper to keep all uniform.

correct that was the reason i wrote it so complicated... :)

Reply:
tested without `.replace("_", " ")` and just upper doesn't seem to remove them :)

Reply:
is it this the method? when i click on github it showed this, maybe they are wrong https://github.com/karrioapi/karrio/blob/5b235d94ca02495a7d3996425653ac9b4b10b56c/modules/core/karrio/server/core/utils.py#L194-L198